app/services/current-route.js

```diff
@@ -0,0 +1,16 @@
+import Ember from 'ember';
+
+const {
+  getOwner,
+  Service,
+  set
+} = Ember;
+
+export default Service.extend({
+  currentRouteName: null,
+
+  updateCurrentRouteName() {
+    let { currentPath } = getOwner(this).lookup('controller:application');
+    set(this, 'currentRouteName', currentPath);
+  }
```

Comment:
Also note that while I do understand any concerns with this service being marked as private right now, there have been RFCs about opening it up, the documentation itself states it, and it seems to be a big part in making components routable, so I would not be too concerned about it.